### ISSUE TITLE
xrefs popup

### ISSUE THREAD

**Issue by psifertex** [Vector35/binaryninja-api#5987]
This is for any customer who really prefers a particular style of xref window -- can we have a modal variant of the Cross References window? Clearly should be a setting (disabled by default).

Or better yet, maybe it's just another variant that can be triggered with an action and no setting is required, just changing the hotkey from "Focus Cross References" to "Modal Cross References" for "X"?

**Comment by psifertex** [comment]
There's two things needed to make this work with the "window" option set on the xrefs window.

1. "Enter" or double clicking to select an item and navigate it needs to close the window when windows.
2. When windowed, "X" doesn't cause the window to appear again.

**Comment by psifertex** [comment]
added in builds >=6233

**Comment by psifertex** [comment]
Re-opening this to track a more complete "modal" dialog.

The current work-around gets us much of the way there, but there are a number of issues associated with it still that make it less convenient.

Any pane that supports navigation events could have an optional "modal" style along with "pane", "floating", and "window". In modal mode, any navigation event would cause the window to close while navigating in the main window and the window itself would block access to the main BN window until navigation happened (to prevent the dialog from getting lost).

**Comment by WeiN76LQh** [comment]
It would be cool if the modal could also do like the peek windows do in VSCode and show the code surrounding the reference so you don't actually have to go to it. I love the peek windows as a great way to look at references to something or remind yourself of somethings definition without actually having to navigate to it.

**Comment by emesare** [comment]
Another idea that I think would be valuable for those working with a large number of cross references, is the ability to display a single regions (section / segment, or otherwise range) xrefs. If you are dealing with a 100000 references you likely do not actually care about 99% of them across your binary, and instead want to see ones that are located within "X". For shared cache this would be an image or a few images.

**Comment by WeiN76LQh** [comment]
I agree with the idea of having more filtering functionality but I'm not sure I agree with your reasoning. Lots of times you're trying to find interesting references to something and what defines "interesting" is very contextual and not necessarily to do with where the reference occurs. For instance I might be looking for calls that provide a certain value for a parameter.

**Comment by emesare** [comment]
> I agree with the idea of having more filtering functionality but I'm not sure I agree with your reasoning. Lots of times you're trying to find interesting references to something and what defines "interesting" is very contextual and not necessarily to do with where the reference occurs. For instance I might be looking for calls that provide a certain value for a parameter.

Right, I am giving a single example for why I would like such a filter, my use case is looking at calls to something that has previously been inlined into a bunch of images that i do not care to see. Right now that workflow is to use the textual search with the region i care about.

I am not saying we don't give the user the option to see all xrefs, I just think we need to find more ways to optionally cut down the number of xrefs that are shown.

**Comment by plafosse** [comment]
We ran out of time to do it in the Helion release. We've got some good ideas on how to move forward.

**Comment by D0ntPanic** [comment]
A true modal dialog option has been added in 5.2.8200

To use the modal dialog, set the setting `ui.defaultXrefInterface` to `dialog`.
